grumpy-runtime-src/runtime/dict.go

```diff
@@ -395,38 +488,52 @@ func (d *Dict) putItem(f *Frame, key, value *Object, overwrite bool) (*Object, *
 	if raised != nil {
 		return nil, raised
 	}
+	hashv := hash.Value()
 	d.mutex.Lock(f)
 	t := d.table
 	v := d.version
-	index, entry, raised := t.lookupEntry(f, hash.Value(), key)
+	index, entry, raised := t.lookupEntry(f, hashv, key)
+	if raised != nil {
+		d.mutex.Unlock(f)
+		return nil, raised
```

Comment:
Can all this `d.mutex.Unlock(f)` before a `return` be replaced by a `defer` right after the `d.mutex.Lock()` on line 491 instead?

Reply:
`defer` is not free. Some time ago it even costed. But probably, putItem is too expensive itself.
I will run benchmark, and if you considered, difference doesn't matter, then I will.

Reply:
As I've said, `defer` is not free:
```
BenchmarkDictSetItem/3-elements-4        658           874           +32.83%
BenchmarkDictSetItem/5-elements-4        1271          1599          +25.81%
BenchmarkDictSetItem/8-elements-4        1609          2153          +33.81%
BenchmarkDictSetItem/12-elements-4       2845          3616          +27.10%
BenchmarkDictSetItem/16-elements-4       3301          4347          +31.69%
BenchmarkDictSetItem/24-elements-4       5673          6999          +23.37%
BenchmarkDictSetItem/32-elements-4       6516          8452          +29.71%
```
Do you really think it has to be in such hot place?

Reply:
Note, that i've tested on i5-5100 , and it is quite old processor today.
Could some one test on more modern one to prove or disprove this proportion?

Reply:
This is odd. I would expect both to be equivalent and act like a macro for the compiler, putting it before the returns for us.

Now I vote for let it be as is. Someday will be better. This day we replace everything with a `defer`.

Can you please add a code comment about this decision, for the next coder?

Reply:
Yes, I did in the next commit.